src/main/java/digital/slovensko/autogram/core/errors/AutogramException.java

```diff
@@ -53,4 +54,16 @@ public static AutogramException createFromDSSException(DSSException e) {
 
         return new UnrecognizedException(e);
     }
+
+    public static AutogramException createFromIllegalArgumentException(IllegalArgumentException e) {
+        for (Throwable cause = e; cause != null && cause.getCause() != cause; cause = cause.getCause()) {
+            if (cause.getMessage() != null) {
+                if (new SigningCertificateExpiredMessagePredicate().test(cause.getMessage())) {
```

Comment:
Rozumiem co robis, ale toto mi pride predcasna abstrakcia na predikat, normalne by som ten REGEX dal ako konstatnu a tuto to otestoval priamo je to oneliner, vyrabat kvoli tomu triedu mi pride zbytocne, specialne ked sa nikde inde nepouzije.